integration-cli/docker_api_containers_test.go

```diff
@@ -684,33 +687,92 @@ func (s *DockerSuite) TestContainerApiCreateWithHostName(c *check.C) {
 	if err != nil && !strings.Contains(err.Error(), "200 OK: 201") {
 		c.Fatal(err)
 	}
-	type createResp struct {
-		Id string
+
+	var container types.ContainerCreateResponse
+	if err := json.Unmarshal(b, &container); err != nil {
+		c.Fatal(err)
 	}
-	var container createResp
+
+	var id = container.ID
+
+	_, bodyGet, err := sockRequest("GET", "/containers/"+id+"/json", nil)
```

Comment:
I understand we just care about the body here but does make sense to check a successful status here?